src/powershell-native/Install-PowerShellRemoting.ps1

```diff
@@ -97,90 +104,119 @@ function Generate-PluginConfigFile
+function Install-PluginEndpoint {
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact="Medium")]
+    param (
+        [Parameter()] [bool] $Force,
+        [switch]
+        $VersionIndependent
+    )
 
-if (! ([Security.Principal.WindowsPrincipal] [Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole] "Administrator"))
-{
-    Write-Error "WinRM registration requires Administrator rights. To run this cmdlet, start PowerShell with the `"Run as administrator`" option."
-    Break
-}
+    ######################
+    #                    #
+    # Install the plugin #
+    #                    #
+    ######################
 
-if ($PsCmdlet.ParameterSetName -eq "ByPath")
-{
-    $targetPsHome = $PowerShellHome
-    $targetPsVersion = & "$targetPsHome\pwsh" -NoProfile -Command '$PSVersionTable.PSVersion.ToString()'
-}
-else
-{
-    ## Get the PSHome and PSVersion using the current powershell instance
-    $targetPsHome = $PSHOME
-    $targetPsVersion = $PSVersionTable.PSVersion.ToString()
-}
+    if ($PsCmdlet.ParameterSetName -eq "ByPath")
+    {
+        $targetPsHome = $PowerShellHome
+        $targetPsVersion = & "$targetPsHome\pwsh" -NoProfile -Command '$PSVersionTable.PSVersion.ToString()'
+    }
+    else
+    {
+        ## Get the PSHome and PSVersion using the current powershell instance
+        $targetPsHome = $PSHOME
+        $targetPsVersion = $PSVersionTable.PSVersion.ToString()
+    }
 
-Write-Verbose "Using PowerShell Version: $targetPsVersion" -Verbose
+    # For default, not tied to the specific version endpoint, we apply
+    # only first number in the PSVersion string to the endpoint name.
+    # Example name: 'PowerShell.6'.
+    if ($VersionIndependent) {
+        $dotPos = $targetPsVersion.IndexOf(".")
+        if ($dotPos -ne -1) {
+            $targetPsVersion = $targetPsVersion.Substring(0, $dotPos)
+        }
+    }
 
-$pluginBasePath = Join-Path "$env:WINDIR\System32\PowerShell" $targetPsVersion
+    Write-Verbose "Using PowerShell Version: $targetPsVersion" -Verbose
 
-$resolvedPluginAbsolutePath = ""
-if (! (Test-Path $pluginBasePath))
-{
-    Write-Verbose "Creating $pluginBasePath"
-    $resolvedPluginAbsolutePath = New-Item -Type Directory -Path $pluginBasePath
-}
-else
-{
-    $resolvedPluginAbsolutePath = Resolve-Path $pluginBasePath
-}
+    $pluginEndpointName = "PowerShell.$targetPsVersion"
 
-$pluginPath = Join-Path $resolvedPluginAbsolutePath "pwrshplugin.dll"
+    $endPoint = Get-PSSessionConfiguration $pluginEndpointName -Force:$Force -ErrorAction silentlycontinue 2>&1
 
-# This is forced to ensure the the file is placed correctly
-Copy-Item $targetPsHome\pwrshplugin.dll $resolvedPluginAbsolutePath -Force -Verbose
+    # If endpoint exists and -Force parameter was not used, the endpoint would not be overwritten.
+    if ($endpoint -and !$Force)
+    {
+        Write-Host "Endpoint $pluginEndpointName already exists." -foregroundcolor Magenta
```

Comment:
This should probably be a `Write-Error` this is what other CmdLets do when you have not specified `-force` and they cannot do what you want.

Reply:
Fixed.